### ISSUE TITLE
Question: Issue with labelIdle and opening file browser

### ISSUE THREAD

**Issue by Jokinen** [pqina/react-filepond#66]
When `labelIdle` is set through the options, and we are using document level event listeners, `filepond--label-action` doesn't consistently open the file browser.

We are using the following component to complete actions when the user clicks outside of a certain component:
```jsx
import React, { Component } from 'react'

export class OutsideClick extends Component {
  selfRef = null

  componentDidMount() {
    document.addEventListener('mousedown', this.handleClickOutside)
  }

  componentWillUnmount() {
    document.removeEventListener('mousedown', this.handleClickOutside)
  }

  setSelfRef = (ref) => this.selfRef = ref

  handleClickOutside = (event) => {
    const { target } = event

    if (
      this.selfRef
      && target instanceof Node
      && !this.selfRef.contains(target)
    ) {
      this.props.onOutsideClick(event)
    }
  }

  render() {
    return (
      <div ref={ this.setSelfRef }>
        { this.props.children }
      </div>
    )
  }
}
```

When we use `<Filepond />` with a custom `labelIdle` close (in the component hierarchy) to the component above, the link for opening the file browser begins to get fidgety. For Safari it wouldn't open at all, for Chrome it would be inconsistent and would fail to properly apply the outline etc.

When I changed from `mousedown` into `click`, `<Filepond />` begun to work as expected.

I wasn't able to narrow down on the problem with more depth than this. It was quite difficult to get this far to be honest 😄 Do you know what could cause the problem here and if there's a more correct way to negotiate the situation? I guess I wanted to ask here to leave a trail in case someone faces a similar issue in the future.

I attempted to create a quick reproduction, but I wasn't able to because I am not sure how `<Filepond />` and our own component need to relate to each other to cause issues. I guess that `<Filepond />` needs to be created only after the `<OutsideClick />` has had its chance to add the listeners, but I wasn't able to get it to not work 😄

**Comment by rikschennink** [comment]
I think your component might be preventing the default event action in the `this.props.onOutsideClick(event)`?

**Comment by Jokinen** [comment]
That's a really good suggestion actually! I double checked and the `onOutsideClick` props do not use the event in any obvious way at least.

**Comment by rikschennink** [comment]
Then without a test case I'm not sure what's going on.

**Comment by Jokinen** [comment]
No worries 👍 

Are you interested about this issue from the perspective of the library? If it'll help you guys get peace of mind (etc), I can try and produce a good test case.

Otherwise, I got the thing working for our purposes so as far as I am concerned it's ok to close this issue. The cause is probably more convoluted than I first suspected as the listeners are a bit hard to keep track of. Maybe someone will stumble on this issue later on with the same problem and we'll get more details as to what could have caused the conflict :)
